package RLGlue;


message RLExperimentDesc {

	required string env_config = 1;
	required string agent_config = 2;

}


message RLStateDesc {

}

message RLActionDesc {

	repeated float action = 1;
	//	repeated float float_action = 1;
	//	repeated int int_action = 2;

}






message RLAgentDesc {

  extensions 1000 to max;

	enum Type {
    DUMMY  = 0;
    SOCKET = 1;
	}

	required Type type = 1;

}

message RLAgentDummyDesc {

  extend RLAgentDesc {
    required RLAgentDummyDesc agent = 1000;
	}

}




message RLEnvironmentCommand {

	enum Type {
    ENV_INIT    = 0;
		ENV_START   = 1;
    ENV_STEP    = 2;
    ENV_CLEANUP = 3;
	}

	required Type type = 1;

	optional InitCommand    initCommand = 2;
	optional StartCommand   startCommand = 3;
	optional StepCommand    stepCommand = 4;
	optional CleanupCommand cleanupCommand = 5;


	message InitCommand {
	}

	message StartCommand {
	}

	message StepCommand {
	}

	message CleanupCommand {
	}

}
